dlt/common/storages/normalize_storage.py

```diff
@@ -47,10 +47,17 @@ def build_extracted_file_stem(schema_name: str, table_name: str, file_id: str) -
     @staticmethod
     def parse_normalize_file_name(file_name: str) -> TParsedNormalizeFileName:
         # parse extracted file name and returns (events found, load id, schema_name)
-        if not file_name.endswith("jsonl"):
+        file_name_p: Path = Path(file_name)
+        if file_name_p.suffix not in (".jsonl", ".parquet"):
             raise ValueError(file_name)
 
-        parts = Path(file_name).stem.split(".")
+        parts = file_name_p.stem.split(".")
         if len(parts) != 3:
             raise ValueError(file_name)
         return TParsedNormalizeFileName(*parts)
+
+    def delete_extracted_files(self, files: Sequence[str]) -> None:
+        for file_name in files:
+            if Path(file_name).suffix == ".parquet":  # Parquet extract files are moved, not copied
```

Comment:
we do not copy those files back into old location right? so we'll have a partially deleted original folder when something fails.

on next attempt files will not be processed again

what we should IMO do is not move but use FileStorage.link_hard to create hard links from the original into temp folder. 

in that case you just delete the original folder like any other.

the only problem I see is when normalize and load storages are on separate filesystem. but in that case rename also falls back to copy so you can use link_hard with fallback to copy. I think it is good compromise